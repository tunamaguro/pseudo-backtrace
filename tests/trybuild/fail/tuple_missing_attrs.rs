use pseudo_backtrace::StackError;

#[derive(Debug, StackError)]
struct TupleMissingAttrs(std::io::Error, &'static core::panic::Location<'static>);

impl core::fmt::Display for TupleMissingAttrs {
    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
        write!(f, "tuple")
    }
}

impl core::error::Error for TupleMissingAttrs {}

fn main() {}
