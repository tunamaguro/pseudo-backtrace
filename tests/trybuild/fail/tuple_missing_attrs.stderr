error: need `#[location]` attribute or field named `location`
 --> tests/trybuild/fail/tuple_missing_attrs.rs:4:8
  |
4 | struct TupleMissingAttrs(std::io::Error, &'static core::panic::Location<'static>);
  |        ^^^^^^^^^^^^^^^^^
